test/acquisition/test_objective.py

```diff
@@ -320,7 +320,7 @@ def test_constrained_mc_objective(self):
                 obj=constrained_obj,
                 constraints=[feasible_con, infeasible_con],
                 samples=samples,
-                infeasible_cost=0.0,
+                infeasible_cost=torch.tensor([0.0], device=self.device, dtype=dtype),
```

Comment:
you're accepting `Union[Tensor, float]` - should we test support both input types?

Reply:
There are something like 6 tests in there. I only changed two of them to use tensors, so both float and tensor are being tested.